src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -42,15 +44,15 @@ class MarathonSchedulerActor(
     taskFailureRepository: TaskFailureRepository,
     config: MarathonConf) extends Actor with ActorLogging with Stash {
   import context.dispatcher
-
   import mesosphere.marathon.MarathonSchedulerActor._
 
-  val appLocks = LockManager[PathId]()
+  var lockedApps = Set.empty[PathId]
   var scheduler: SchedulerActions = _
-
   var deploymentManager: ActorRef = _
   var historyActor: ActorRef = _
 
+  val cancellationTimeout: FiniteDuration = 1.minute
```

Comment:
Not sure, this is more a security mechanism to prevent the actor from staying in the `awaitCancellation` state when there's a bug. Not really something the user should care about.